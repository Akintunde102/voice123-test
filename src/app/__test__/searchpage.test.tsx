import { render, screen } from "@testing-library/react";
import SearchPage from '../search/page'

describe('SearchPage', () => {

    it('renders SearchPage component with input with correct placeholder', () => {
        render(<SearchPage />)

        const input = screen.getByPlaceholderText(/Cartoon Voice Actor/i)

        expect(input).toBeInTheDocument()
    });



    it("renders searchPage unchanged", () => {
        const { container } = render(<SearchPage />);
        expect(container).toMatchSnapshot();
    });

    it('renders no result text when there is no result', () => {
        render(<SearchPage />)

        const noResultText = screen.getByText(/loading data/i)

        expect(noResultText).toBeInTheDocument()
    });


})

