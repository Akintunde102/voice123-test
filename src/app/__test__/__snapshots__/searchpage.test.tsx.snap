// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage renders searchPage unchanged 1`] = `
<div>
  <div
    class="MuiBox-root css-e6j8w8"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-149yiww-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1b8q9yj"
      >
        <form>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-0.5 css-1hwevs-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9 MuiGrid-grid-md-10 css-1h7x1lk-MuiGrid-root"
            >
              <div
                aria-label="Search for Voice Actors"
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-rpemjh-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-hiddenLabel css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    aria-label="Search for Voice Actors"
                    autocomplete="on"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputHiddenLabel css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="outlined-basic"
                    placeholder="Cartoon Voice Actor"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3 MuiGrid-grid-md-2 css-7li7ay-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-572x3s-MuiButtonBase-root-MuiButton-root"
                disabled=""
                tabindex="-1"
                type="submit"
              >
                <span
                  class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="SearchIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                    />
                  </svg>
                </span>
                Search
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
    <span
      class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
      role="progressbar"
    >
      <span
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
      />
      <span
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
      />
    </span>
    <div
      class="MuiBox-root css-7f9tzs"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-d59r90-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-19g0w5n-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          >
            ...Loading data for
          </h6>
          <span
            class="MuiTypography-root MuiTypography-subtitle1 css-hlkk1l-MuiTypography-root"
          >
               "
            "
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
